torchdynamo/variables/misc.py

```diff
@@ -249,6 +249,33 @@ def call_function(
             return WrappedUserFunctionVariable(args[0], self)
 
 
+class SimpleEnterExitContextWrappingVariable(ContextWrappingVariable):
+    def __init__(self, cm, **kwargs):
+        super(SimpleEnterExitContextWrappingVariable, self).__init__(
+            target_values=True, initial_values=False, **kwargs
+        )
+        self.cm = cm
+
+    def enter(self, tx):
+        def enter_functional():
+            return self.cm.__enter__()
+
+        tx.output.graph.create_node("call_function", enter_functional, (), {})
+        return variables.ConstantVariable(None, **VariableTracker.propagate(self))
```

Comment:
This is unsound, enter does not always return None.

Reply:
Agreed. Same as below, I'll narrow it down to a list of specific managers.

Reply:
RM